docs/migration_guides/qi_migration.rst

```diff
@@ -0,0 +1,699 @@
+################################
+Quantum Instance Migration Guide
+################################
+
+The :class:`~qiskit.utils.QuantumInstance` is a utility class that allowed to jointly
+configure the circuit transpilation and execution steps, and provided useful tools for algorithm development,
+such as basic error mitigation strategies.
+
+The functionality of :class:`~qiskit.utils.QuantumInstance.execute` has
+now been delegated to the different implementations of the :mod:`~qiskit.primitives` base classes,
+while the explicit transpilation has been left to the :mod:`~qiskit.transpiler` module (see table below).
+Thus, the :class:`~qiskit.utils.QuantumInstance` is being deprecated.
+
+Summary of migration alternatives for the :class:`~qiskit.utils.QuantumInstance` class:
+
+.. list-table::
+   :header-rows: 1
+
+   * - QuantumInstance method
+     - Alternative
+   * - ``QuantumInstance.execute``
+     - ``Sampler.run`` or ``Estimator.run``
```

Comment:
We're probably going to deprecate `execute`, so that would not need to be listed as alternative 🙂 See also https://github.com/Qiskit/qiskit-terra/pull/7892